docs/integration/DataTypingAndSchemasHandling.md

```diff
@@ -0,0 +1,246 @@
+| Any matching type from the list of types in the union schema | union                                    | Read more about [validation modes](#validation-and-encoding).                                                     |
+
+If at runtime value cannot be converted to an appropriate logic schema (e.g. "notAUUID" cannot be converted to
+proper UUID), then an error will be reported.
+
+## JSON Schema
+
+We support [JSON Schema](https://json-schema.org/) in version: `Draft 7` without:
+
+[//]: # (See SwaggerBasedJsonSchemaTypeDefinitionExtractor)
+
+* Numbers with a zero fractional part (e.g. `1.0`) as a proper value on decoding (deserialization)
+  for [integer schema](https://json-schema.org/understanding-json-schema/reference/numeric.html#integer)
+* Nested relative schemas
+* Recursive schemas
+* Anchors
+
+JSON Schema is available on [Streaming](/docs/scenarios_authoring/DataSourcesAndSinks.md)
+and [Request - Response](/docs/scenarios_authoring/RRDataSourcesAndSinks.md). To integrate with JSON on
+streaming we use [Schema Registry](/docs/integration/KafkaIntegration.md#schema-registry-integration). On the
+other hand, we have Request - Response where [schemas](/docs/scenarios_authoring/RRDataSourcesAndSinks.md) are stored in
+scenario properties.
+
+### Source conversion mapping
+
+| JSON Schema                                                                                        | Java type  | Comment                                                                                                                                                            |
+|----------------------------------------------------------------------------------------------------|------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [null](https://json-schema.org/understanding-json-schema/reference/null.html)                      | null       |                                                                                                                                                                    |
+| [string](https://json-schema.org/understanding-json-schema/reference/string.html)                  | String     | UTF-8                                                                                                                                                              |
+| [boolean](https://json-schema.org/understanding-json-schema/reference/boolean.html)                | Boolean    |                                                                                                                                                                    |
+| [integer](https://json-schema.org/understanding-json-schema/reference/numeric.html#integer)        | Long       | 64bit, TODO: Support for [range](https://json-schema.org/understanding-json-schema/reference/numeric.html#range) and do proper conversion to: Int/Long/BigInteger. |
```

Comment:
Let's leave TODO for the code...

Reply:
Hmm, I have different opinion, but I will remove it.